### PR TITLE
Add grpc_posix bindings.

### PR DESCRIPTION
**Pull request by ced5**
This will allow Python users to use POSIX file descriptors to create
gRPC channels. Both client and server bindings are included.

The test, unfortunately, hangs. I haven't yet figured out why.

I'm also concerned about platforms where the `grpc_posix` bindings
aren't available (*cough*Windows*cough*). The test shouldn't run
there.

### CODE REVIEW

File: src/python/grpcio_tests/tests/unit/_channel_fd_test.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
+# Copyright 2016 gRPC authors.
```
Comment (nathanielmanistaatgoogle) [inline src/python/grpcio_tests/tests/unit/_channel_fd_test.py]: I suspect this line was copied and pasted?

Comment (ced5) [reply src/python/grpcio_tests/tests/unit/_channel_fd_test.py]: Yep. I've updated it now.

**Comment by nathanielmanistaatgoogle** [comment]
@ced5: how are the tests for this coming along?

**Comment by yashykt** [comment]
@lidizheng please close if no longer relevant

**Comment by lidizheng** [comment]
@ced5 Can you please provide more context about this feature? Since it is adding public API, I would like to know the demands of this feature. If it is no longer needed, please close this pull request.

**Pull request closed without merging**